scripts/01_Seurat_QC.Rmd

```diff
@@ -616,47 +326,56 @@ plot_dr <- data.frame(X = paste0(reduction_names, "_1"),
   
     print(current_plot <- plot_scatter(metadata = metadata,
-                               out_path = file.path(out_path, "dr"),
-                               proj_name = sample_name,
-                               log_file = log_file,
-                               X = plot_dr[i,1],
-                               Y = plot_dr[i,2],
-                               color = "nFeature_RNA",
-                               write = TRUE))
+                                       scratch_dir,
+                                       proj_name = sample_name,
+                                       log_file = log_file,
+                                       X = plot_dr[i,1],
+                                       Y = plot_dr[i,2],
+                                       color = "nFeature_RNA",
+                                       write = TRUE))
     
     print(current_plot <- plot_scatter(metadata = metadata,
-                           out_path = file.path(out_path, "dr"),
-                           proj_name = sample_name,
-                           log_file = log_file,
-                           X = plot_dr[i,1],
-                           Y = plot_dr[i,2],
-                           color = "nCount_RNA",
-                           write = TRUE))
+                                       scratch_dir,
+                                       proj_name = sample_name,
+                                       log_file = log_file,
+                                       X = plot_dr[i,1],
+                                       Y = plot_dr[i,2],
+                                       color = "nCount_RNA",
+                                       write = TRUE))
     
     print(current_plot <- plot_scatter(metadata = metadata,
-                       out_path = file.path(out_path, "dr"),
-                       proj_name = sample_name,
-                       log_file = log_file,
-                       X = plot_dr[i,1],
-                       Y = plot_dr[i,2],
-                       color = "pct_mito",
-                       write = TRUE))
+                                       scratch_dir,
+                                       proj_name = sample_name,
+                                       log_file = log_file,
+                                       X = plot_dr[i,1],
+                                       Y = plot_dr[i,2],
+                                       color = "pct_mito",
+                                       write = TRUE))
 }
 ```
 
-We will save the filtered seurat object to be used for further downstream analysis ("seurat_obj.rds").
+# Save filtered metadata and Seurat object
+
+We will save the filtered seurat object to be used for further downstream analysis (`seurat_obj.rds`).
+
+```{r save-seurat-obj, echo=TRUE}
+# Save metadata
+write_tsv(metadata, path = glue("{results_dir}/metadata_create_{sample_name}.tsv"))
+
+# Save Seurat object
+saveRDS(seurat_obj, file = paste0(results_dir, "/", "seurat_obj.rds"))
```

Comment:
we should probably have the option to give these unique names. perhaps a `output_basename` param.